diff --git a/node_modules/@types/parse-torrent/index.d.ts b/node_modules/@types/parse-torrent/index.d.ts
index 12bd2d7..c68c055 100644
--- a/node_modules/@types/parse-torrent/index.d.ts
+++ b/node_modules/@types/parse-torrent/index.d.ts
@@ -7,9 +7,9 @@ declare const ParseTorrent: ParseTorrent.ParseTorrent;
 
 declare namespace ParseTorrent {
     interface ParseTorrent {
-        (torrent: string): MagnetUri.Instance;
-        (torrent: Buffer): MagnetUri.Instance | ParseTorrentFile.Instance;
-        (torrent: Instance | MagnetUri.Instance | ParseTorrentFile.Instance): Instance;
+        (torrent: string): Promise<MagnetUri.Instance>;
+        (torrent: Buffer): Promise<MagnetUri.Instance | ParseTorrentFile.Instance>;
+        (torrent: Instance | MagnetUri.Instance | ParseTorrentFile.Instance): Promise<Instance>;
 
         toMagnetURI: typeof MagnetUri.encode;
         toTorrentFile: typeof ParseTorrentFile.encode;
