diff --git a/node_modules/@types/stremio-addon-sdk/index.d.ts b/node_modules/@types/stremio-addon-sdk/index.d.ts
index bf10aa1..c91179d 100644
--- a/node_modules/@types/stremio-addon-sdk/index.d.ts
+++ b/node_modules/@types/stremio-addon-sdk/index.d.ts
@@ -398,22 +398,39 @@ export interface Stream {
          */
         notWebReady?: boolean | undefined;
         /**
-         * If defined, addons with the same behaviorHints.group will be chosen automatically for binge watching.
+         * If defined, addons with the same `behaviorHints.bingeGroup` will be chosen automatically for binge watching.
          * 
-         * This should be something that identifies the stream's nature within your addon.
-         * For example, if your addon is called "gobsAddon", and the stream is 720p, the group should be "gobsAddon-720p".
-         * If the next episode has a stream with the same group, stremio should select that stream implicitly.
+         * This should be something that identifies the stream's nature within your addon: for example, 
+         * if your addon is called "gobsAddon", and the stream is 720p, the bingeGroup should be "gobsAddon-720p".
+         * If the next episode has a stream with the same bingeGroup, stremio should select that stream implicitly.
          */
-        group?: string | undefined;
+        bingeGroup?: string | undefined;
         /**
-         * **Not implemented yet!**
+         * **Only applies to urls.**
          * 
-         * HTTP headers to use when trying to playback url.
-         * Only applies to urls.
-         *
-         * @ignore
+         * When using this property, you must also set `stream.behaviorHints.notWebReady: true`
+         * This is an object containing request and response which include the headers that should
+         * be used for the stream (example value: `{ "request": { "User-Agent": "Stremio" } }`).
+         */
+        proxyHeaders?: Record<string, string>;
+        /**
+         * The calculated OpenSubtitles hash of the video, this will be used when the streaming
+         * server is not connected (so the hash cannot be calculated locally), this value is passed
+         * to subtitle addons to identify correct subtitles.
+         */
+        videoHash?: string;
+        /**
+         * Size of the video file in bytes, this value is passed to the subtitle addons to identify correct subtitles.
+         */
+        videoSize?: number;
+        /**
+         * Filename of the video file, although optional, it is highly recommended to set it when 
+         * using `stream.url` (when possible) in order to identify correct 
+         * subtitles (addon sdk will show a warning if it is not set in this case).
+         * This value is passed to the subtitle addons to identify correct subtitles.
          */
-        headers?: any;
+        filename?: string;
+        
     } | undefined;
 }
 
