diff --git a/node_modules/@types/stremio-addon-sdk/index.d.ts b/node_modules/@types/stremio-addon-sdk/index.d.ts
index 7f1a205..4bbf892 100644
--- a/node_modules/@types/stremio-addon-sdk/index.d.ts
+++ b/node_modules/@types/stremio-addon-sdk/index.d.ts
@@ -380,6 +380,7 @@ export interface Stream {
      * Usually a short name to identify the addon that provided the stream
      */
     name?: string | undefined;
+    description?: string | undefined;
     /**
      * Array of Subtitle objects representing subtitles for this stream.
      */
@@ -516,6 +517,20 @@ export interface Manifest {
          * Used to provide an adequate warning to the user.
          */
         p2p?: boolean | undefined;
+        /**
+         * If the addon supports settings, will add a button next to "Install" in Stremio 
+         * that will point to the /configure path on the addon's domain.
+         * 
+         * For more information read [User Data](https://github.com/Stremio/stremio-addon-sdk/blob/master/docs/api/responses/manifest.md#user-data) (or if you are not using the Addon SDK, read: [Advanced User Data](https://github.com/Stremio/stremio-addon-sdk/blob/master/docs/advanced.md#using-user-data-in-addons) and [Creating Addon Configuration Pages](https://github.com/Stremio/stremio-addon-sdk/blob/master/docs/advanced.md#creating-addon-configuration-pages))
+         */
+        configurable?: boolean | undefined;
+        /**
+         * If set to true the "Install" button will not show for your addon in Stremio, 
+         * instead a "Configure" button will show pointing to the /configure path on the addon's domain.
+         * 
+         * For more information read [User Data](https://github.com/Stremio/stremio-addon-sdk/blob/master/docs/api/responses/manifest.md#user-data) (or if you are not using the Addon SDK, read: [Advanced User Data](https://github.com/Stremio/stremio-addon-sdk/blob/master/docs/advanced.md#using-user-data-in-addons) and [Creating Addon Configuration Pages](https://github.com/Stremio/stremio-addon-sdk/blob/master/docs/advanced.md#creating-addon-configuration-pages))
+         */
+        configurationRequired?: boolean | undefined;
     } | undefined;
 }
 
